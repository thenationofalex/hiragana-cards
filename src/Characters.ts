import {CONSTANTS} from './Constants';

const {SPACER} = CONSTANTS;
export interface ICharacters {
  ro: string;
  jp: string;
}

export const Characters: ICharacters[] = [
  {ro: 'a', jp: 'あ'},
  {ro: 'i', jp: 'い'},
  {ro: 'u', jp: 'う'},
  {ro: 'e', jp: 'え'},
  {ro: 'o', jp: 'お'},
  {ro: 'ka', jp: 'か'},
  {ro: 'ki', jp: 'き'},
  {ro: 'ku', jp: 'く'},
  {ro: 'ke', jp: 'け'},
  {ro: 'ko', jp: 'け'},
  {ro: 'sa', jp: 'さ'},
  {ro: 'shi', jp: 'し'},
  {ro: 'su', jp: 'す'},
  {ro: 'se', jp: 'せ'},
  {ro: 'so', jp: 'そ'},
  {ro: 'ta', jp: 'た'},
  {ro: 'chi', jp: 'ち'},
  {ro: 'tsu', jp: 'つ'},
  {ro: 'te', jp: 'て'},
  {ro: 'to', jp: 'と'},
  {ro: 'na', jp: 'と'},
  {ro: 'ni', jp: 'に'},
  {ro: 'nu', jp: 'ぬ'},
  {ro: 'ne', jp: 'ね'},
  {ro: 'no', jp: 'の'},
  {ro: 'ha', jp: 'は'},
  {ro: 'hi', jp: 'ひ'},
  {ro: 'fu', jp: 'ふ'},
  {ro: 'he', jp: 'へ'},
  {ro: 'ho', jp: 'ほ'},
  {ro: 'ma', jp: 'ま'},
  {ro: 'mi', jp: 'み'},
  {ro: 'mu', jp: 'む'},
  {ro: 'me', jp: 'め'},
  {ro: 'mo', jp: 'も'},
  {ro: 'ya', jp: 'や'},
  {ro: SPACER, jp: `${SPACER}-1`},
  {ro: 'yu', jp: 'ゆ'},
  {ro: SPACER, jp: `${SPACER}-2`},
  {ro: 'yo', jp: 'よ'},
  {ro: 'ra', jp: 'ら'},
  {ro: 'ri', jp: 'り'},
  {ro: 'ru', jp: 'る'},
  {ro: 're', jp: 'れ'},
  {ro: 'ro', jp: 'ろ'},
  {ro: 'wa', jp: 'わ'},
  {ro: 'wi', jp: 'ゐ'},
  {ro: SPACER, jp: `${SPACER}-3`},
  {ro: 'we', jp: 'ゑ'},
  {ro: 'wo', jp: 'を'},
  {ro: 'n', jp: 'ん'},
];
