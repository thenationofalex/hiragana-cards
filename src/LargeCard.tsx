import {
  Dimensions,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import FlipCard from 'react-native-flip-card';
import React from 'react';

import {ICharacters} from './Characters';
import {hasIn} from 'lodash';

interface ILargeCard {
  char: ICharacters;
  setTableInView: (arg0: boolean) => void;
}

const {height: screenHeight, width: screenWidth} = Dimensions.get('screen');

const LargeCard = ({char, setTableInView}: ILargeCard) => {
  const combinedCardTextStyle = [styles.allText, styles.cardText];
  const cardActionTextStyle = [styles.allText, styles.cardActionText];

  return (
    <SafeAreaView>
      <View style={styles.container}>
        <View style={styles.cardContainer}>
          <FlipCard>
            <View style={styles.card}>
              <Text style={combinedCardTextStyle}>{char.jp}</Text>
            </View>
            <View style={styles.card}>
              <Text style={[...combinedCardTextStyle, styles.cardTextSmall]}>
                {char.jp}
              </Text>
              <Text style={combinedCardTextStyle}>{char.ro}</Text>
            </View>
          </FlipCard>
        </View>
        <View style={styles.cardActionBar}>
          <TouchableOpacity onPress={() => setTableInView(true)}>
            <Text style={cardActionTextStyle}>ALL CARDS</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => setTableInView(true)}>
            <Text style={cardActionTextStyle}>NEXT CARD</Text>
          </TouchableOpacity>
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'orange',
    display: 'flex',
    height: screenHeight,
  },
  card: {
    justifyContent: 'center',
    display: 'flex',
    height: screenHeight * 0.8,
    width: screenWidth,
  },
  cardContainer: {
    display: 'flex',
    height: screenHeight * 0.8,
    width: screenWidth,
  },
  cardText: {
    color: 'white',
    fontSize: screenWidth / 2,
  },
  cardTextSmall: {
    fontSize: screenWidth / 8,
    opacity: 0.5,
  },
  allText: {
    textAlign: 'center',
  },
  cardActionBar: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  cardActionText: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 18,
    marginHorizontal: 20,
    borderWidth: 1,
    padding: 10,
    borderColor: 'transparent',
    borderRadius: 10,
    overflow: 'hidden',
    backgroundColor: 'rgba(0, 0, 0, 0.1)',
  },
});

export default LargeCard;
