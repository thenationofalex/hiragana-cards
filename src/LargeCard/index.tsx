import {
  Dimensions,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import FlipCard from 'react-native-flip-card';
import React from 'react';

import {ICharacters} from '../Characters';

interface ILargeCard {
  char: ICharacters;
  setNextCharacterInView: (arg0: boolean) => void;
  setTableInView: (arg0: boolean) => void;
}

const {height: screenHeight, width: screenWidth} = Dimensions.get('screen');

const LargeCard = ({
  char,
  setTableInView,
  setNextCharacterInView,
}: ILargeCard) => {
  const cardActionTextStyle = [styles.allText, styles.cardActionText];
  const combinedCardTextStyle = [styles.allText, styles.cardText];
  const tapTextStyle = [styles.allText, styles.tapTextStyles];

  return (
    <SafeAreaView>
      <View style={styles.container}>
        <View style={styles.tapStyle}>
          <Text style={tapTextStyle}>Tap character to flip card</Text>
        </View>
        <View style={styles.cardContainer}>
          <FlipCard>
            <View style={styles.card}>
              <Text style={combinedCardTextStyle}>{char.jp}</Text>
            </View>
            <View style={styles.card}>
              <Text style={[...combinedCardTextStyle, styles.cardTextSmall]}>
                {char.jp}
              </Text>
              <Text style={combinedCardTextStyle}>{char.ro}</Text>
            </View>
          </FlipCard>
        </View>
        <View style={styles.cardActionBar}>
          <TouchableOpacity onPress={() => setTableInView(true)}>
            <Text style={cardActionTextStyle}>ALL CARDS</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => setNextCharacterInView(false)}>
            <Text style={cardActionTextStyle}>NEXT CARD</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => setNextCharacterInView(true)}>
            <Text style={cardActionTextStyle}>RANDOM CARD</Text>
          </TouchableOpacity>
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'orange',
    display: 'flex',
    height: screenHeight,
  },
  card: {
    justifyContent: 'center',
    display: 'flex',
    height: screenHeight * 0.8,
    width: screenWidth,
  },
  cardContainer: {
    display: 'flex',
    height: screenHeight * 0.8,
    width: screenWidth,
  },
  cardText: {
    color: 'white',
    fontSize: screenWidth / 2,
  },
  cardTextSmall: {
    fontSize: screenWidth / 8,
    opacity: 0.5,
  },
  allText: {
    textAlign: 'center',
  },
  cardActionBar: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  cardActionText: {
    backgroundColor: 'rgba(0, 0, 0, 0.1)',
    borderColor: 'transparent',
    borderRadius: 10,
    borderWidth: 1,
    color: 'white',
    fontSize: 14,
    fontWeight: 'bold',
    marginHorizontal: 10,
    overflow: 'hidden',
    padding: 10,
  },
  tapStyle: {
    position: 'absolute',
    textAlign: 'center',
    opacity: 0.5,
    width: '100%',
  },
  tapTextStyles: {
    color: 'white',
    fontWeight: 'bold',
    marginTop: 32,
  },
});

export default LargeCard;
