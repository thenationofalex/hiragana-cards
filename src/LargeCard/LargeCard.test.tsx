import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';

import {Characters} from '../Characters';
import LargeCard from '.';

const props = {
  char: Characters[0],
  setNextCharacterInView: () => jest.fn(),
  setTableInView: () => jest.fn(),
};

describe('LargeCard', () => {
  it('Renders correctly', () => {
    const app = render(<LargeCard {...props} />);
    expect(app).toMatchSnapshot();
  });
});
