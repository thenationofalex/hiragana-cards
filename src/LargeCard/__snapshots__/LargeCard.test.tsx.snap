// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeCard Renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "backgroundColor": "orange",
        "display": "flex",
        "height": 1334,
      }
    }
  >
    <View
      style={
        Object {
          "opacity": 0.5,
          "position": "absolute",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginTop": 32,
            },
          ]
        }
      >
        Tap character to flip card
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "height": 1067.2,
          "width": 750,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          alignHeight={false}
          alignWidth={false}
          clickable={true}
          flip={false}
          flipHorizontal={false}
          flipVertical={true}
          friction={6}
          onFlipEnd={[Function]}
          onFlipStart={[Function]}
          perspective={1000}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
              "transform": Array [
                Object {
                  "perspective": 1000,
                },
                Object {
                  "rotateX": "0deg",
                },
              ],
            }
          }
          useNativeDriver={true}
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Array [
                  false,
                  false,
                ],
              ]
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "height": 1067.2,
                  "justifyContent": "center",
                  "width": 750,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "textAlign": "center",
                    },
                    Object {
                      "color": "white",
                      "fontSize": 375,
                    },
                  ]
                }
              >
                „ÅÇ
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.1)",
                "borderColor": "transparent",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "overflow": "hidden",
                "padding": 10,
              },
            ]
          }
        >
          ALL CARDS
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.1)",
                "borderColor": "transparent",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "overflow": "hidden",
                "padding": 10,
              },
            ]
          }
        >
          NEXT CARD
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.1)",
                "borderColor": "transparent",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
                "marginHorizontal": 10,
                "overflow": "hidden",
                "padding": 10,
              },
            ]
          }
        >
          RANDOM CARD
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
